Mayavi example crashes sphinx build - (think it's not tested by Travis)

Hello sphinx-gallery people,

this issue might be not related to sphinx-gallery (but to mayavi instead), so feel free to close it if you think it's not relevant for the project.

I'm using sphinx-gallery master (24a11317ccf8), and I tried to build the doc running ``make html`` inside the ``docs`` folder. 

The build works fine until the mayavi example, that crashes the build
```
plotting code blocks in ../mayavi_examples/plot_3d.py
The program 'sphinx-build' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadValue (integer parameter out of range for operation)'.
  (Details: serial 55 error_code 2 request_code 154 minor_code 3)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the --sync command line
   option to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
Makefile:62: recipe for target 'html' failed
make: *** [html] Error 1
```
It might be a problem with my installation of Mayavi (but I tried both with mayavi 4.3 packaged by Ubuntu and with mayavi master 4.4.4dev) , since closing a Mayavi figure inside of IPython results in a segmentation fault. Adding 
```
from mayavi.core.engine import Engine
from mayavi import mlab

e = Engine()
e.start()
mlab.set_engine(e)
```
to the script solves the problem for me, since the build is not crashing any more. 

Also, it seems that Travis is not testing the build of mayavi gallery examples (I might be wrong, but a quick search through Travis logs convinced me that it is not tested). This example might be skipped on purpose, but since .travis.yml says "# This environment tests is for mayavi" (https://github.com/sphinx-gallery/sphinx-gallery/blob/master/.travis.yml#L18), I was a bit puzzled.

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,6 +7,7 @@ addons:
     - python-matplotlib
     - python-pip
     - python-coverage
+    - xvfb # For mayavi headless
 
 virtualenv:
         system_site_packages: true
@@ -38,6 +39,9 @@ install:
         if [ "$PYTHON_VERSION" == "2.7" ]; then
           conda install --yes --quiet mayavi;
           conda upgrade --yes --all;
+          # virtual framebuffer for mayavi rendering
+          Xvfb :1 -screen 0 1280x1024x24 -auth localhost &
+          export DISPLAY=:1
         fi;
       fi;
     - pip install -r requirements.txt

--- a/.travis.yml
+++ b/.travis.yml
@@ -34,10 +34,11 @@ install:
     - if [ "$DISTRIB" == "conda" ]; then
         conda create --yes -n testenv python=$PYTHON_VERSION pip numpy
         setuptools matplotlib pillow sphinx nose coverage;
+        source activate testenv;
         if [ "$PYTHON_VERSION" == "2.7" ]; then
           conda install --yes --quiet mayavi;
+          conda upgrade --yes --all;
         fi;
-        source activate testenv;
       fi;
     - pip install -r requirements.txt
     - python setup.py install

--- a/.travis.yml
+++ b/.travis.yml
@@ -39,9 +39,8 @@ install:
         if [ "$PYTHON_VERSION" == "2.7" ]; then
           conda install --yes --quiet mayavi;
           conda upgrade --yes --all;
-          # virtual framebuffer for mayavi rendering
-          Xvfb :1 -screen 0 1280x1024x24 -auth localhost &
-          export DISPLAY=:1
+          Xvfb :1 -screen 0 1280x1024x24 -auth localhost &;
+          export DISPLAY=:1;
         fi;
       fi;
     - pip install -r requirements.txt

--- a/.travis.yml
+++ b/.travis.yml
@@ -39,6 +39,7 @@ install:
         if [ "$PYTHON_VERSION" == "2.7" ]; then
           conda install --yes --quiet mayavi;
           conda upgrade --yes --all;
+          conda upgrade --yes pyface;
         fi;
       fi;
     - pip install -r requirements.txt

--- a/.travis.yml
+++ b/.travis.yml
@@ -39,12 +39,17 @@ install:
         if [ "$PYTHON_VERSION" == "2.7" ]; then
           conda install --yes --quiet mayavi;
           conda upgrade --yes --all;
-          Xvfb :1 -screen 0 1280x1024x24 -auth localhost &;
-          export DISPLAY=:1;
         fi;
       fi;
     - pip install -r requirements.txt
     - python setup.py install
+
+# To test the mayavi environmen follow the instructions at
+# https://docs.travis-ci.com/user/gui-and-headless-browsers/#Using-xvfb-to-Run-Tests-That-Require-a-GUI
+before_script:
+  - "export DISPLAY=:99.0"
+  - "sh -e /etc/init.d/xvfb start"
+  - sleep 3 # give xvfb some time to start
 
 script:
     - if [ "$DISTRIB" == "ubuntu" ]; then python setup.py nosetests; fi

